### PR TITLE
chore: Reduce release yaml

### PR DESCRIPTION
**Pull request by tico24**
### Motivation

I noticed some duplicate code in the release yaml that could be compacted down.

### Verification

Ran the release a number of times on our fork.

(If you're reviewing all my commits you'll see that I experimented with improving the caching. Due to the way the docker container is constructed, more agressive caching results in failed builds, so I have reverted those changes.)

### CODE REVIEW

**Review by caelan-io — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -81,110 +81,10 @@ jobs:
...
-          tag_suffix=$(echo $PLATFORM | sed -r "s/\//-/g")
-          image_name="${DOCKERIO_ORG}/${TARGET}:${tag}-${tag_suffix}"
-
-          docker buildx build \
-            --cache-from "type=local,src=/tmp/.buildx-cache" \
-            --cache-to "type=local,dest=/tmp/.buildx-cache" \
-            --output "type=image,push=true" \
-            --build-arg GIT_COMMIT=$GIT_COMMIT \
-            --build-arg GIT_TAG=$GIT_TAG \
-            --build-arg GIT_TREE_STATE=$GIT_TREE_STATE \
```
Comment (terrytangyuan) [inline .github/workflows/release.yaml:172]: Why did we remove these?

Comment (tico24) [reply .github/workflows/release.yaml:172]: over-enthustic delete caused by looking at the 3.4.8 branch. They are back now.

**Review by terrytangyuan — APPROVED** [review]
Could you run another test release before I merge this?

**Comment by tico24** [comment]
> Could you run another test release before I merge this?

my test release had those lines in. I just failed to cherry pick the right lines across from one repo to another.

**Comment by terrytangyuan** [comment]
Commits seem messed up. Mind cleaning up a bit?

**Comment by tico24** [comment]
I have no idea what happened there, but I'm enjoying the fact that this PR has nearly the same number of commits as LOC changed.

**Review by terrytangyuan — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -20,102 +20,19 @@ permissions:
...
-        platform: [ linux/arm64 ]
+        platform: [ linux/amd64, linux/arm64 ]
         target: [ workflow-controller, argocli, argoexec ]
     steps:
       - uses: actions/checkout@v3
 
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2
-        with:
-          platforms: arm64
```
Comment (terrytangyuan) [inline .github/workflows/release.yaml:118]: Just saw this change. Could you confirm that the built images work as expected for multi-arch?

Comment (tico24) [reply .github/workflows/release.yaml:118]: yes, if you don't set it, qemu and buildkit just do the right thing.

Comment (tico24) [reply .github/workflows/release.yaml:118]: and here's the screenshot from quay showing the multi-arch image
image

Comment (terrytangyuan) [reply .github/workflows/release.yaml:118]: Great thanks!

**Pull request merged**